spec/cictl/index_command_spec.rb

```diff
@@ -26,23 +26,15 @@
       expect(solr_ids("deleted:true")).to include(bogus_delete)
     end
 
-    it "bails out if redirects file can't be found" do
-      # Would be nice if we could class_double Services, but alas that would require
-      # patching various parameters to :[] and there are plenty of them.
-      @save_redirect_file = HathiTrust::Services[:redirect_file]
-      @save_no_redirects = HathiTrust::Services[:no_redirects?]
-      @save_no_external_data = HathiTrust::Services[:no_external_data?]
-      @save_redirects = HathiTrust::Services[:redirects]
-      HathiTrust::Services.register(:redirect_file) { "no_such_redirects_file.gz.txt" }
-      HathiTrust::Services.register(:no_redirects?) { false }
-      HathiTrust::Services.register(:no_external_data?) { false }
-      expect {
-        CICTL::Commands.start(["index", "all", "--no-wait", "--quiet", "--log", test_log])
-      }.to raise_error(CICTL::CICTLError)
-      HathiTrust::Services.register(:redirect_file) { @save_redirect_file }
-      HathiTrust::Services.register(:no_redirects?) { @save_no_redirects }
-      HathiTrust::Services.register(:no_external_data?) { @save_no_external_data }
-      HathiTrust::Services.register(:redirects) { @save_redirects }
+    context "using nonexistent redirect file" do
+      override_service(:redirect_file) { "no_such_redirects_file.gz.txt" }
```

Comment:
`redirects` itself should get re-composed after the dependent `redirect_file` gets updated. So long as `redirect_file` itself doesn't throw an exception, I don't think this should cause any stack issues.